subsys/bootloader/bl_crypto/bl_crypto_ed25519.c

```diff
@@ -54,9 +55,25 @@ int bl_ed25519_validate(const uint8_t *data, uint32_t data_len, const uint8_t *s
 		LOG_DBG("PSA verify message result: %d", status);
 
 		if (status == PSA_SUCCESS) {
-			return 0;
+			verified_with = i;
+			break;
 		}
 	}
+#if defined(CONFIG_SB_CRYPTO_KMU_KEYS_REVOCATION)
+	for (int i = 0; i < verified_with; ++i) {
+		psa_key_id_t kid = kmu_key_ids[i];
+
+		status = psa_destroy_key(kid);
+		if (status == PSA_SUCCESS) {
+			LOG_DBG("Success invalidating key ID %d", i);
+		} else {
+			LOG_ERR("Key invalidation failed with: %d", status);
```

Comment:
imagine this should infinite loop?

Reply:
that's signed comparison. modified it to int32 in case we have a lots of keys :D